app/hotels/src/allHotels/AllHotels.js

```diff
@@ -17,53 +15,69 @@ type Props = {|
   onFilterChange: (filter: SearchParametersType) => void,
 |};
 
-export default class AllHotelsSearch extends React.Component<Props> {
+export default class AllHotels extends React.Component<Props> {
+  /**
+   * Submit form with the initial date interval parameters.
+   */
+  componentDidMount = () => {
+    this.props.onFilterChange({
+      checkin: moment()
+        .add(1, 'week')
+        .startOf('isoWeek')
+        .format('YYYY-MM-DD'),
+      checkout: moment()
+        .add(1, 'week')
+        .endOf('isoWeek')
+        .format('YYYY-MM-DD'),
+    });
+  };
+
   handleSearchChange = (search: SearchParametersType) => {
     this.props.onFilterChange(search);
   };
 
-  isReadyToSearch = (): boolean => {
-    const { search: s } = this.props;
+  renderInnerComponent = rendererProps => {
+    const {
+      onFilterChange,
+      openSingleHotel,
+      search,
+      location,
+      onLocationChange,
+    } = this.props;
+    const data = idx(rendererProps, _ => _.props.city) || null;
+
     return (
-      s.latitude !== null &&
-      s.longitude !== null &&
-      s.checkin !== null &&
-      s.checkout !== null
+      <AllHotelsSearch
+        data={data}
+        search={search}
+        location={location}
+        onFilterChange={onFilterChange}
+        onLocationChange={onLocationChange}
+        openSingleHotel={openSingleHotel}
+      />
     );
   };
 
-  renderInnerComponent = (propsFromRenderer: AllHotelsSearchQueryResponse) => (
-    <AllHotelsSearchList
-      data={propsFromRenderer.allHotels}
-      openSingleHotel={this.props.openSingleHotel}
-    />
-  );
-
   render = () => {
-    const { search } = this.props;
+    const { location } = this.props;
+
     return (
-      <Layout>
-        <SearchForm onChange={this.handleSearchChange} />
-        <FilterStripe />
-        {this.isReadyToSearch() && (
-          <PublicApiRenderer
-            query={graphql`
-              query AllHotelsSearchQuery($search: HotelsSearchInput!) {
-                allHotels: allAvailableHotels(search: $search) {
-                  ...AllHotelsSearchList
-                }
-              }
-            `}
-            variables={{
-              search: search,
-            }}
-            render={this.renderInnerComponent}
-            cacheConfig={{
-              force: true,
-            }}
-          />
-        )}
-      </Layout>
+      <SimpleQueryRenderer
```

Comment:
Oh, now I understand. I was confused by the `simple` name. So it's like a "background" query renderer? It should be used only when you need to fetch something at the background (location in this case) and you don't want to show all the unnecessary spinners and errors, right?

So I would add it to the `AllHotelsSearch` or maybe directly to the `SearchForm`. It seems weird that you need to introduce two additional props just to use this query renderer in the parent component and send the data back down. It's very confusing.

Also, does it make sense to introduce new query like this?

```graphql
{
  allHotels: allAvailableHotelsInCity(city, checkin, checkout, roomsConfiguration) {
    ...AllHotelsSearchList
  }
}
```

It would be even nicer this way because you don't need to send the city ID...

Reply:
I can imagine that

The only possible drawback  scenario I can imagine is to show something like "you entered silly city name, try again something different"

Reply:
Now it behaves totally weird, no loading indicator (for quite a while to be honest), blank page (I tried "Pilsen" and it's blank forever - is it an error, still loading?), to short city = no feedback. Make it work, please.

Reply:
This issue should be solved now